src/App.tsx

```diff
@@ -1,61 +1,95 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { Todo } from './types/todo';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [user, setUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [todos, setTodos] = useState(todosFromServer);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(user <= 0);
+
+    if (!title || user <= 0) {
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    const newTodo: Todo = {
+      id: maxId + 1,
+      title: title,
+      completed: false,
+      userId: user,
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUser(0);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
```

Comment:
According to React standartds it's better to divide all into components. So here will be better to create component for Form and use it. Because all form logic in App.tsx - its not good at all;